README.md

```diff
@@ -1 +1,278 @@
+
+`check` is a macro that runs our tests. Between `check all` and `do`, we define our generators, which generate the input for our tests. Here we're using [`StreamData.string`](https://hexdocs.pm/stream_data/StreamData.html#string/2) and [`StreamData.list_of`](https://hexdocs.pm/stream_data/StreamData.html#list_of/2) to create a random length list of random length strings. Because these functions are imported within `property`, we don't need to prefix them with the `StreamData` module name.
+
+We're passing options to our generators here. We pass `:printable` to string, this means our generator will create strings using all printable unicode characters. We also pass `min_length: 1` to `list_of` to ensure we aren't given an empty list. See [the docs](https://hexdocs.pm/stream_data/StreamData.html#content) for a full list of options for all generators.
+
+Then, inside the body (between `do` and `end`) we make our assertions. These assertions should be as simple as possible. If your properties are too complex, you would need to write tests to validate those, and those tests would need to be validated, and so on... Limiting our assertions to be as simple as possible means we can look at them and instantly be sure they are correct.
+
+Now we'll define our other properties
+
+``` elixir
+property "Frame width = length of longest input string + 4" do
+    check all l <- string(:printable) |> list_of(min_length: 1) do
+      frame_width = l |> Frame.build() |> List.first() |> String.length()
+      longest_string = Enum.max_by(l, &String.length/1) |> String.length()
+
+      assert frame_width == longest_string + 4
+    end
+  end
+
+property "All sides are equal width" do
+    check all l <- string(:printable) |> list_of(min_length: 1) do
+      frame_width = l |> Frame.build() |> List.first() |> String.length()
+
+      assert Enum.all?(Frame.build(l), fn s -> String.length(s) == frame_width end)
+    end
+  end
+```
+
+As you can see, we're using the same generator for all of these properties, so we can factor that out into a function:
+
+``` elixir
+defp list_string() do
+  string(:printable) |> list_of(min_length: 1)
+end
+
+property "height = length of list + 2" do
+  check all l <- list_string() do
+    ...
+  end
+end
+
+property "Frame width = length of longest input string + 4" do
+  check all l <- list_string() do
+    ...
+  end
+end
+
+property "All sides are equal width" do
+  check all l <- list_string() do
+    ...
+  end
+end
+```
+
+Now that we have our properties defined, let's write some code to test.
+
+``` elixir
+defmodule Frame do
+  def build(input) do
+    longest = Enum.max_by(input, fn s -> String.length(s) end) |> String.length()
+
+    top = String.duplicate("*", longest + 4)
+
+    input
+    |> Enum.map(fn s ->
+      "* " <> s <> suffix(longest - String.length(s) + 1)
+    end)
+    |> List.insert_at(0, top)
+    |> List.insert_at(-1, top)
```

Comment:
I think I would have used `&String.pad_trailing/2` for this

```elixir
[border]
|> Enum.concat(Enum.map(strings, fn s -> "* " <> String.pad_trailing(s, longest - 4) <> " *") end)
|> Enum.concat([border])
```

(and remove the `suffix` function)